### PR TITLE
Fix/improve types. Upgrade amqplib to version 0.8.0. Bump to version 1.8.0

### PR DESCRIPTION
**Pull request by jesperborgstrup**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -57,15 +57,15 @@ export interface Event {
   eventName: string;
   context: any;
   uuid: string;
-  time: Date;
+  time: number;
```
Comment (jeppenejsum) [inline index.d.ts:60]: Why this change?

**Review by mbnissen — APPROVED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ interface RetryFixedBackoffConfiguration {
   delay?: number
 }
 
-interface RetryConfiguration {
+type RetryConfiguration = false | {
```
Comment (jeppenejsum) [inline index.d.ts:85]: This seems a bit backwards imo. Wouldn't it make more sense to not provide a RetryConfiguration if retry is not needed?

Comment (jesperborgstrup) [reply index.d.ts:85]: This is merely reflecting the types in the implementation (and the README)

File: index.d.ts (hunk 1 above)
Comment (jesperborgstrup) [reply index.d.ts:60]: The types were wrong and didn't match the implementation. Timestamps are milliseconds since unix epoch, not JS Date objects.
